core/contracts/financial-templates/implementation/Liquidatable.sol

```diff
@@ -95,32 +95,23 @@ contract Liquidatable is PricelessPositionManager {
     // TODO: add more fields to this event after refactoring the withdrawn function
     event LiquidationWithdrawn(address caller);
 
-    // TODO: could this modifier be replaced with one called `onlyPreDispute` and then the function can use
-    // the `onlyPreExpiration` modifier from the base contract and this one in conjunction?
-
-    // Callable before the liquidation's expiry AND there is no pending dispute on the liquidation
-    modifier onlyPreExpiryAndPreDispute(uint id, address sponsor) {
+    // Callable if the liquidation is in a state where it can be disputed.
+    modifier disputable(uint id, address sponsor) {
         LiquidationData storage liquidation = _getLiquidationData(sponsor, id);
         require((getCurrentTime() < liquidation.expiry) && (liquidation.state == Status.PreDispute));
         _;
     }
-    // Callable either post the liquidation's expiry or after a dispute has been resolved,
-    // i.e. once a dispute has been requested, the liquidation's expiry ceases to matter
-    modifier onlyPostExpiryOrPostDispute(uint id, address sponsor) {
+    // Callable if the liquidation is in a state where someone can withdraw.
+    modifier withdrawable(uint id, address sponsor) {
         LiquidationData storage liquidation = _getLiquidationData(sponsor, id);
         Status state = liquidation.state;
+
+        // Must be disputed or the liquidation has passed expiry.
         require(
-            (state == Status.DisputeSucceeded) ||
-                (state == Status.DisputeFailed) ||
-                ((liquidation.expiry <= getCurrentTime()) && (state == Status.PreDispute))
+            (state > Status.PreDispute) || ((liquidation.expiry <= getCurrentTime()) && (state == Status.PreDispute))
```

Comment:
`state > Status.PreDispute` I like this, but maybe make a note to keep the states in "chronological" order

Reply:
So to be clear, you have essentially added an OR condition here `(state == Status.PendingDispute)` because in `PendingDispute` we want the `_settle` internal method to run

Reply:
Yep, that is correct. Agreed, I'll add a comment to keep them in chronological order.

Reply:
Done.